### ISSUE TITLE
Why do detection models report APval only and what is reason behind of not considering test?

### ISSUE THREAD

**Issue by optimisticabouthumanity** [ultralytics/ultralytics#15986]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear everyone,
why detections models are evaluated only on validation set?
Why test set is not necessary?

@Ultralytics team, would you please also share your insight/knowledge about it?

**Comment by Skillnoob** [comment]
You can use the test set to compare different models that you have trained, without any bias from the val set.

**Comment by optimisticabouthumanity** [comment]
> You can use the test set to compare different models that you have trained, without any bias from the val set.

Hi, look at the papers of YOLOv7, YOLOv9, RT-DETR. All of them reported APval. Why do not see any APtest? What could be the reason?

**Comment by Y-T-G** [comment]
You usually just report the results on one split. There's not need to report the results on both validation and test because it's redundant.

And detection models usually use COCO `val2017` split as benchmark since the labels for that is public. The COCO test set labels are not public. They only allow you to upload the results on their website to measure the performance on test set.

**Comment by optimisticabouthumanity** [comment]
> You usually just report the results on one split. There's not need to report the results on both validation and test because it's redundant.
> 
> And detection models usually use COCO `val2017` split as benchmark since the labels for that is public. The COCO test set labels are not public. They only allow you to upload the results on their website to measure the performance on test set.

Hi, thank for your informative reply.
Would you please share why is it redundant? Can a model overfit on both training and validation set? If not, test set can be redundant. Otherwise, isn't test set crucial to actually know how the model is doing?
If I train on custom dataset, should I keep a separate test set from total data? Or is it redundant also?

**Comment by Y-T-G** [comment]
The `val2017` set is used to validate, because it has become the de-facto standard in benchmarks. There's no point in providing performance on the test set because very few papers use it and the primary purpose of the results is to compare the performance of your model with others. If no one else has provided the test set results, you can't compare it with theirs. And the labels are also not public like I said, so you can't also show the visualizations of how your model performed vs. the ground truth.

During training, the model only learns from the training set. The validation set is not learnt from. A test set is simply a good practice because sometimes you may optimize your model on the validation set by tuning the hyperparameters. Test set becomes important in this scenario to make sure the hyperparameters isn't only good on the validation set. Otherwise, you don't really need test set.
